src/Stratis.Bitcoin.Features.SignalR/EventSubscriptionService.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Microsoft.Extensions.Logging;
+using Stratis.Bitcoin.EventBus;
+using Stratis.Bitcoin.Signals;
+
+namespace Stratis.Bitcoin.Features.SignalR
+{
+    public class EventSubscriptionService : IEventsSubscriptionService, IDisposable
+    {
+        private readonly SignalROptions options;
+        private readonly ISignals signals;
+        private readonly EventsHub eventsHub;
+        private readonly ILogger<SignalRFeature> logger;
+        private readonly List<SubscriptionToken> subscriptions = new List<SubscriptionToken>();
+
+        public EventSubscriptionService(
+            SignalROptions options,
+            ILoggerFactory loggerFactory,
+            ISignals signals,
+            EventsHub eventsHub)
+        {
+            this.options = options;
+            this.signals = signals;
+            this.eventsHub = eventsHub;
+            this.logger = loggerFactory.CreateLogger<SignalRFeature>();
+        }
+
+        public void Init()
+        {
+            MethodInfo subscribeMethod = this.signals.GetType().GetMethod("Subscribe");
+            MethodInfo onEventCallbackMethod = typeof(EventSubscriptionService).GetMethod("OnEvent");
+            foreach (IClientEvent eventToHandle in this.options.EventsToHandle)
+            {
+                this.logger.LogDebug("Create subscription for {eventType}", eventToHandle.NodeEventType);
```

Comment:
We don't use string interpolation for logging messages.

Reply:
Done

Reply:
> We don't use string interpolation for logging messages.

This is not a string interpolation (note no $). This is a native new feature for NLog that also supports structured logging.

Reply:
https://github.com/NLog/NLog/wiki/How-to-use-structured-logging